src/interpreter/fvm.rs

```diff
@@ -218,24 +227,47 @@ impl<DB: Blockstore> Consensus for ForestExternsV2<DB> {
                 // check blocks are properly signed by their respective miner
                 // note we do not need to check extra's: it is a parent to block b
                 // which itself is signed, so it was willingly included by the miner
-                if let Ok(gas_used) = self.verify_block_signature(&bh_1) {
-                    total_gas += gas_used;
-                    if let Ok(gas_used) = self.verify_block_signature(&bh_2) {
+                let res = self.verify_block_signature(&bh_1);
+
+                let match_error = |err: Error, total_gas: i64| {
+                    match err {
+                        // When a lookup error occurs we should just bail terminating all the
+                        // computations.
+                        Error::Lookup(_) => {
+                            error!("database lookup error: {}", err);
+                            Err(err)
+                        }
+                        // invalid consensus fault: cannot verify block header signature
+                        Error::Signature(_) => Ok((None, total_gas)),
```

Comment:
Do we know why is it okay to discard this error? I get that this logic was there before, I'm just curious.

Reply:
Yes, AFAIK when the signature is not correct - we ignore the consensus fault, but we need to report the total gas used for the lookup.